lumen/sources/base.py

```diff
@@ -93,6 +93,24 @@ def _filter_dataframe(cls, df, **query):
             df = df[mask]
         return df
 
+    @classmethod
+    def _resolve_reference(cls, reference, sources={}):
+        refs = reference[1:].split('.')
+        if len(refs) != 3:
+            raise ValueError(f"Reference string '{reference} not valid. "
+                             "Must take the form @source.table.field.")
+        sourceref, table, field = refs
+        source = cls.from_spec(sourceref, sources)
+        table_schema = source.get_schema(table)
```

Comment:
Can getting a table not fail? Seems like you're only checking for field and enum not being found, but I guess that's because a table not found will just result in an empty set of results (and thus a missing field)? Seems like the field error message might confuse people in that case, as it indicates that the field was not found in the table but doesn't say anything about whether the table itself was found.